src/Documents/DocumentStore.ts

```diff
@@ -85,14 +85,42 @@ export class DocumentStore implements IDocumentStore {
       .then((): IDocumentStore => this);
   }
 
-  public openSession(database?: string, forceReadFromMaster: boolean = false): IDocumentSession {
+  public async openSession(transaction: ITransaction) : Promise<void>;
+  public async openSession(options: ISessionOptions, transaction: ITransaction) : Promise<void>;
+  public openSession(database?: string, forceReadFromMaster?: boolean) : IDocumentSession;
+  public openSession(firstArg?: string | ITransaction | ISessionOptions, secondArg?: ITransaction | boolean): IDocumentSession | Promise<void> {
```

Comment:
@ayende probably i can leave it just as optional feature ? now you can use in both ways
i already updated all unit test for use  load/store wrapped in callbacks - if we remove callback - i need to rewrite all them again (
also, anyway we  need to dispose used Hilo ranges. in Python it's done via __exit__() method which calls after end of with(store) ... block. In js there are with operator, but, no specific "magic" methods for it and also it's obsolette and non-reccomended for use